src/core/csv/reader_fread.cc

```diff
@@ -754,30 +754,57 @@ void FreadReader::detect_header() {
 //------------------------------------------------------------------------------
 
 /**
- * This helper method tests whether '\\n' characters are present in the file,
- * and sets the `cr_is_newline` flag accordingly.
- *
- * If '\\n' exists in the file, then `cr_is_newline` is false, and standalone
- * '\\r' will be treated as a regular character. However if there are no '\\n's
- * in the file (at least within the first 100 lines), then we will treat '\\r'
- * as a newline character.
- */
+  * This method attempts to detect the value of the `cr_is_newline`
+  * parse parameter. When `cr_is_newline` is true, a standalone '\\r'
+  * character is treated as a line separator. When the value is false,
+  * a standalone '\\r' does not start a new line.
+  *
+  * Only the first 64Kb of text will be scanned. If any \\n is found,
+  * then cr_is_newline will be set to false. However, if at least 10
+  * \\r are found without encountering a single \\n, then
+  * cr_is_newline will be set to true. If after reading 64Kb of input
+  * only \\r's were seen, then cr_is_newline will again be set to
+  * true. If neither \\r's nor \\n's were seen, then cr_is_newline
+  * will be set to false (i.e. newlines are \\n-based).
```

Comment:
What about creating python options for `64Kb` and `10`?

Reply:
If the need ever arises, it is much easier to expose the `eol` parameter to fread.
Because, what we're really trying to detect here are files that use `\r` as a line separator, which doesn't happen very often. According to Wikipedia (https://en.wikipedia.org/wiki/Newline), this was used for "Commodore 8-bit machines (C64, C128), Acorn BBC, ZX Spectrum, TRS-80, Apple II series, Oberon, the classic Mac OS, MIT Lisp Machine and OS-9".  While apparently some csv files still remain from those times, it is highly unlikely that they would have line lengths > 64Kb.

Going in the opposite direction, we're running the risk that a `\n`-using file will have an unquoted field with 10+ `\r`s in the very first line (which has to be the header line usually). Again, incredibly unlikely, and the authors of the file are supposed to quote those fields anyways.

Reply:
Yes, so why not to expose the eol parameter to fread? I was like it is already exposed and on this PR we just improve auto-detection.

Reply:
We should, it's just not within the scope of this PR. Another parameter that we also want exposed is the `quote_rule`.